src/pruna/evaluation/evaluation_agent.py

```diff
@@ -33,12 +38,44 @@ class EvaluationAgent:
 
     Parameters
     ----------
-    task : Task
+    task : Task, optional
         Configuration object that defines how to evaluate the model.
+        Deprecated: Use metrics, datamodule, and device parameters directly.
+    metrics : str | List[str | BaseMetric | StatefulMetric], optional
+        The metrics to evaluate. Required if task is not provided.
+    datamodule : PrunaDataModule, optional
+        The dataloader to use for the evaluation. Required if task is not provided.
+    device : str | torch.device | None, optional
+        The device to be used, e.g., 'cuda' or 'cpu'. Default is None.
+        If None, the best available device will be used.
     """
 
-    def __init__(self, task: Task) -> None:
-        self.task = task
+    def __init__(
+        self,
+        task: Task | None = None,
+        *,
+        metrics: str | List[str | BaseMetric | StatefulMetric] | None = None,
+        datamodule: PrunaDataModule | None = None,
+        device: str | torch.device | None = None,
+    ) -> None:
+        if task is not None:
+            if metrics is not None or datamodule is not None or device is not None:
+                raise ValueError(
+                    "Cannot specify both 'task' parameter and direct parameters (metrics, datamodule, device). "
+                    "Use either the deprecated 'task' parameter or the new direct parameters."
+                )
+            warn(
+                "The 'task' parameter is deprecated and will be removed in a future version. "
```

Comment:
Would you be able to point out a specific release, this canbe 0.2.8
additionally, could you add tests like `test_deprecation_*` etc.